### PR TITLE
Schedule-related improvements

### PR DESCRIPTION
**Pull request by bouwew**
I realized that the thermostat-select for the legacy Anna only has one schedule-option.
The Off-option was added so the select of the Plugwise HA integration works the same as for the non-legacy thermostats.
Also, detection of an empty schedule was added.

Finally, a small correction was made for actual devices.

Furthermore, all `in [..., ...]` constructs were replaced by `in (..., ...)` constructs including one typing-item in constants.py: `Final[list[str]]` to `Final[tuple[str, ...]]`.
And, some double used code was replaced by a common function, `common_match_cases()`.

### CODE REVIEW

File: plugwise/data.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def _device_data_climate(self, device: DeviceData, data: DeviceData) -> None:
         # Operation modes: auto, heat, heat_cool, cool and off
         data["mode"] = "auto"
         self._count += 1
-        if sel_schedule == NONE:
+        if sel_schedule in (NONE, OFF):
```
Comment (CoMPaTech) [inline plugwise/data.py:207]: I would suggest we keep using `lists` (i.e. `[]`) and not `set`s (i.e. `()`)

File: plugwise/legacy/data.py [hunk 2]
```diff
@@ -88,5 +88,5 @@ def _device_data_climate(self, device: DeviceData, data: DeviceData) -> None:
         # Operation modes: auto, heat
         data["mode"] = "auto"
         self._count += 1
-        if sel_schedule == NONE:
+        if sel_schedule in (NONE, OFF):
```
Comment (CoMPaTech) [inline plugwise/legacy/data.py:91]: see earlier comment

**Review by CoMPaTech — APPROVED** [review]

File: plugwise/data.py (hunk 1 above)
Comment (bouwew) [reply plugwise/data.py:207]: See my comment on Discord.

Comment (bouwew) [reply plugwise/data.py:207]: Further changes implemented as indicated on Discord.

**Review by CoMPaTech — APPROVED** [review]

**Pull request merged**